ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -631,9 +623,17 @@ const NetworksForm = ({
             token_symbol: ticker,
           },
         });
+      }
 
-        submitCallback?.();
+      if (addNewNetwork && !setActiveOnSubmit) {
```

Comment:
If `setActiveOnSubmit` is true we switch to the network immediately so we should not show a modal prompting the user to switch to the newly added network (as happens when `setNewNetworkAdded` action is called.